packages/jsii-pacmak/lib/targets/golang/types/interface.ts

```diff
@@ -3,61 +3,121 @@ import { InterfaceType, Method, Property } from 'jsii-reflect';
 import { GoType } from './go-type';
 import { GoTypeRef } from './go-type-reference';
 import { Package } from '../package';
+import { TypeField } from './type-field';
+import { getFieldDependencies } from '../util';
+
+class InterfaceProperty implements TypeField {
+  public readonly name: string;
+  public readonly references?: GoTypeRef;
+
+  public constructor(
+    public readonly parent: Interface,
+    private readonly property: Property,
+  ) {
+    this.name = property.name;
+
+    if (property.type) {
+      this.references = new GoTypeRef(parent.parent.root, property.type);
+    }
+  }
+
+  public emit(code: CodeMaker) {
+    const propName = code.toPascalCase(this.name);
+    const type = new GoTypeRef(
+      this.parent.parent.root,
+      this.property.type,
+    ).scopedName(this.parent.parent);
+
+    code.line(`Get${propName}() ${type}`);
+  }
+}
+
+class InterfaceMethod implements TypeField {
+  public readonly name: string;
+  public readonly references?: GoTypeRef;
+
+  public constructor(
+    public readonly parent: Interface,
+    private readonly method: Method,
+  ) {
+    this.name = method.name;
+
+    if (method.returns.type) {
+      this.references = new GoTypeRef(parent.parent.root, method.returns.type);
+    }
+  }
+
+  public emit(code: CodeMaker) {
+    const returns = this.method.returns.type.void
+      ? ''
+      : ` ${new GoTypeRef(
+          this.parent.parent.root,
+          this.method.returns.type,
+        ).scopedName(this.parent.parent)}`;
+
+    const methodName = code.toPascalCase(this.name);
+
+    code.line(`${methodName}()${returns}`);
+  }
+}
 
 export class Interface extends GoType {
-  public readonly dependencies: Package[] = [];
+  public readonly methods: InterfaceMethod[];
+  public readonly properties: InterfaceProperty[];
 
   public constructor(parent: Package, public type: InterfaceType) {
     super(parent, type);
+    this.methods = Object.values(type.getMethods()).map(
+      (method) => new InterfaceMethod(this, method),
+    );
+    this.properties = Object.values(type.getProperties()).map(
+      (prop) => new InterfaceProperty(this, prop),
+    );
   }
 
   public emit(code: CodeMaker): void {
-    code.line('// Behaviorial interface'); // FIXME for debugging
```

Comment:
I'd like to keep these comments for now to better troubleshoot the generated code.